src/store/reducer/user.js

```diff
@@ -66,6 +66,18 @@ function rootReducer(state = defaultState, action) {
         subject: 'All',
         class: 'All',
       };
+    case 'USERS':
+      const newArray8 = state.all.filter(
+        user => user.username == action.username
```

Comment:
You are defining newArray8 so better to shift it below newArray7.
Also make the filter case in-sensitive
use `===`